### PR TITLE
Fixes #529 Create Form Service 

### LINKED ISSUE #529: Create Form Constants Service
Create a service that holds all constant types for Form and related components.

### PR DESCRIPTION
**Pull request by oliverrayreed**
I created a service for the form elements.

### CODE REVIEW

**Comment by tabersays** [comment]
CI is reporting failing lint issues:
ERROR: /home/rof/src/github.com/buildcom/BossyUI/src/app/form.service.ts[13, 2]: file should end with a newline
ERROR: /home/rof/src/github.com/buildcom/BossyUI/src/app/form.service.ts[4, 25]: missing whitespace
ERROR: /home/rof/src/github.com/buildcom/BossyUI/src/app/form.service.ts[9, 26]: Missing semicolon
ERROR: /home/rof/src/github.com/buildcom/BossyUI/src/app/form.service.ts[10, 25]: Missing semicolon

**Comment by oliverrayreed** [comment]
Do I need to change anything on mine?

File: src/app/form.service.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    TYPE: string = undefined;
+    ISSPLIT = false;
+    ISDROPUP = false;
+    ISRIGHTALIGNED = false;
+    VARIANT = 'Secondary';
+    SIZE: string = undefined;
+    NAME = 'MissingName';
+    HREF = '#';
+    HASVALIDATION = true;
+    DANGER = 'Danger';
```
Comment (glawson) [inline src/app/form.service.ts]: all lower case for Danger, Warning, Error - please.

File: src/app/form.service.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class FormConfigService {
+    TYPE: string = undefined;
+    ISSPLIT = false;
+    ISDROPUP = false;
+    ISRIGHTALIGNED = false;
+    VARIANT = 'Secondary';
```
Comment (glawson) [inline src/app/form.service.ts]: value should be lowercase

File: src/app/form.service.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class FormConfigService {
```
Comment (glawson) [inline src/app/form.service.ts]: rename class to FormConstantService

File: src/app/form.service.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    TYPE: string = undefined;
```
Comment (glawson) [inline src/app/form.service.ts]: Not a constant (enum like) if it's undefined, is this needed?

File: src/app/form.service.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    ISSPLIT = false;
```
Comment (glawson) [inline src/app/form.service.ts]: bool's don't belong in a constant definition class.

**Comment by glawson** [comment]
closes #529

**Pull request merged**